build.gradle

```diff
@@ -139,6 +166,72 @@ allprojects {
     }
 }
 
+def log4jVersion = '2.22.1'
+
+configurations {
+    distOnly
+}
+
+dependencies {
+    distOnly "org.apache.logging.log4j:log4j-core:${log4jVersion}"
+    distOnly "org.apache.logging.log4j:log4j-slf4j2-impl:${log4jVersion}"
+}
```

Comment:
What is `distOnly` configurations used for here? I didn't find any document related to `distOnly`. It seems to be "Adding logging backend jar to `dist` task only". Removing `configurations.distOnly` seems to have no change.

In the project's root dependency, adding specific dependencies seems to contradict `dependencyManagement`. The approach you've provided currently is more of a subtraction followed by addition, which means excluding all logging backends from the project's root directory first, then adding them back to the `dist` and executable classes. As far as I know, most projects adopt a more traditional approach, which is addition followed by subtraction, meaning first introducing logging backends normally, then excluding them from the SDK.

The former approach of "adding logging backends to `dist` and executable classes" overly emphasizes the presence of a specific logging backend and disperses the dependencies of logging backends in multiple places, including `eventmesh-starter`, `eventmesh-connectors`, `dist`, etc. The latter, in the step of "introducing logging backends normally", only requires adding the logging backend the project is using once in `eventmesh-common`.

Personally, I tend to be more conservative and might prefer the latter approach. However, if the former could have a more understandable solution (especially for newcomers to the community), I would also choose the former.